sdk/tests/deqp/framework/common/tcuImageCompare.js

```diff
@@ -279,8 +281,8 @@ tcuImageCompare.floatUlpThresholdCompare = function(imageSetName, imageSetDesc,
 
                 // Instead of memcpy(), which is the way to do float->uint32 reinterpretation in C++
                 for (var i = 0; i < refPix.length; i++) {
-                    refBits[i] = refPix[i];
-                    cmpBits[i] = cmpPix[i];
+                    refBits[i] = tcuFloat.convertFloat32Inline(refPix[i], tcuFloat.description32);
+                    cmpBits[i] = tcuFloat.convertFloat32Inline(cmpPix[i], tcuFloat.description32);
```

Comment:
Previous method cann't handle comparison between -1 and -0.9999999. Result of tcuFloat.convertFloat32Inline can be used to do float number comparison.